src/sentry/utils/kafka.py

```diff
@@ -33,3 +37,128 @@ def get(self, key):
+            "group.id": consumer_group,
+            "enable.auto.commit": "false",  # we commit manually
+            "enable.auto.offset.store": "true",  # we let the broker keep count of the current offset (when committing)
+            "enable.partition.eof": "false",  # stop EOF errors when we read all messages in the topic
+            "default.topic.config": {"auto.offset.reset": initial_offset_reset},
+        }
+
+        if consumer_configuration is not None:
+            for key in six.iterkeys(consumer_configuration):
+                self.consumer_configuration[key] = consumer_configuration[key]
+
+    @abc.abstractmethod
+    def process_message(self, message):
+        """
+        This function is called for each message
+        :param message: the kafka message:
+        """
+        pass
+
+    def run(self, is_shutdown_requested=lambda: False):
+        """
+        Runs the message processing loop
+        """
+        logger.debug(
+            "Staring kafka consumer for topic:{} with consumer group:{}",
+            self.topic_name,
+            self.consumer_group,
+        )
+
+        consumer = kafka.Consumer(self.consumer_configuration)
+        consumer.subscribe([self.topic_name])
+
+        # setup a flag to mark termination signals received, see below why we use an array
+        termination_signal_received = [False]
+
+        def termination_signal_handler(_sig_id, _frame):
+            """
+            Function to use a hook for SIGINT and SIGTERM
+
+            This signal handler only remembers that the signal was emitted.
+            The batch processing loop detects that the signal was emitted
+            and stops once the whole batch is processed.
+            """
+            # We need to use an array so that terminal_signal_received is not a
+            # local variable assignment, but a lookup in the clojure's outer scope.
+            termination_signal_received[0] = True
+
+        with set_termination_request_handlers(termination_signal_handler):
+            while not (is_shutdown_requested() or termination_signal_received[0]):
+                # get up to commit_batch_size messages
+                messages = consumer.consume(
+                    num_messages=self.commit_batch_size, timeout=self.max_fetch_time_seconds
+                )
+
+                for message in messages:
+                    message_error = message.error()
+                    if message_error is not None:
+                        logger.error(
+                            "Received message with error on %s, error:'%s'",
+                            self.topic_name,
+                            message_error,
+                        )
+                        raise ValueError(
+                            "Bad message received from consumer", self.topic_name, message_error
+                        )
+
+                    try:
+                        self.process_message(message)
```

Comment:
Please use `sentry.utils.safe_execute` here. This will run a callback in a try-catch block and log the error appropriately.